kafka-connector/src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceConnector.java

```diff
@@ -71,10 +68,8 @@ public String toString() {
     }
 
     public static PartitionScheme getEnum(String value) {
-      if (value.equals("round_robin")) {
```

Comment:
I don't think it's safe to just get rid of this mode. Other may already be depending on it. We can add new modes, but we shouldn't eliminate old ones, at least as long as we don't have the released code versioned.

Reply:
Anyone who uses `round_robin` now should be able to get the same behaviour with  `kafka_producer` scheme. (Assuming they don't pass message keys and use `round_robin` at the same time)

Reply:
But what if they have round_robin specified in their config? Do they have to switch to kafka_producer? If so, I'd prefer round_robin stay as a mode for now so we get backwards compatibility with configs.

Reply:
So I'll keep all current partition schemes with no changes and add a new `kafka_producer` scheme. Also the `round_robin` will remain default for the same reasons. BTW: I think I have to recreate the PR since I messed up with commit emails a bit.